### PR TITLE
Enable interactive brush and drop down selection for the bar chart

### PR DESCRIPTION
**Pull request by LujieZ**
This PR enables an interactive brush and add a drop down menu 
to allow selections between all pushes and complete ones.

### CODE REVIEW

**Comment by LujieZ** [comment]
Running version here: https://pr32-c36ca11-dot-step189-2020.uc.r.appspot.com/

File: web/src/app/components/bar-chart/bar-chart.component.html [hunk 1]
```diff
@@ -1,7 +1,10 @@
-<div id='instruction'>
-  <h4>Bar Chart</h4>
-</div>
+<!-- Initialize a select button. -->
+<div #selectButton class='selectButton'>
+  <select id='selections' (change)='updateChart();'>
+    <option value='0'>Show all pushes</option>
+    <option value='1'>Show complete pushes</option>
```
Comment (razvanm) [inline web/src/app/components/bar-chart/bar-chart.component.html:5]:
```suggestion
    <option value='1'>Show completed pushes</option>
```

Comment (razvanm) [inline web/src/app/components/bar-chart/bar-chart.component.html:5]: How about using more descriptive values like 'all' and 'completed'?

File: web/src/app/components/bar-chart/bar-chart.component.ts [hunk 2]
```diff
@@ -229,6 +228,11 @@ export class BarChartComponent implements AfterViewInit {
       .attr('transform', `translate(${marginFocus.left}, 0)`)
       .style('color', BarChartComponent.COLOR_LIGHT_GRAY);
 
+    this.focus.append('text')
+      .attr('text-anchor', 'middle')
+      .attr('transform', `translate(${elementWidth / 2}, ${marginFocus.top / 2})`)
+      .style('font-size', '16px sans-serif')
+      .text('Bar chart of all pushes');
```
Comment (yinggao) [inline web/src/app/components/bar-chart/bar-chart.component.ts:235]: It should be 'Bar chart of completed pushes' if a user select to show just completed pushes.

One way is to change the title based on the selection, another easier way is to just call it 'Bar chart for push durations'

File: web/src/app/components/bar-chart/bar-chart.component.html [hunk 3]
```diff
@@ -1,7 +1,10 @@
...
+  </select>
+ </div>
```
Comment (yinggao) [inline web/src/app/components/bar-chart/bar-chart.component.html:7]:
```suggestion
</div>
```

File: web/src/app/components/bar-chart/bar-chart.component.ts [hunk 4]
```diff
@@ -256,8 +256,8 @@ export class BarChartComponent implements AfterViewInit {
    * This function updates the focus bar chart and the brush bar chart based on
    * the data of the dropdown selection. The function display the most recent 30
    * pushes by default. It also implements an interactive brush to display a selected
-   * area of the bar chart. If the selected value is '0', the function updates both
-   * charts with all pushes; if the selected value is '1', it udpates the charts with
+   * area of the bar chart. If the selected value is 'all', the function updates both
+   * charts with all pushes; if the selected value is 'completed', it udpates the charts with
```
Comment (razvanm) [inline web/src/app/components/bar-chart/bar-chart.component.ts:260]: Please reflow.

**Pull request merged**